### PR TITLE
Added StarWars story for Data

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Added StarWars story for Data.

This is to help explore and use Data with a variety of REST API endpoints.

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

no

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

**Comment by MikeKingdom** [comment]
Pretty cool and nice implementation. The API is pretty slow. I think the search needs a debounce. Does the API support a sort?

I did run into a CORS complaint from the browser while doing a couple searches. Not sure why since the same call seemed to work just before.

File: src/js/components/Data/stories/StarWars.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+export const StarWars = () => {
+  const [result, setResult] = useState({});
+  const [view, setView] = useState({ search: '' });
+  const abortRef = useRef();
+
+  useEffect(() => {
+    // This API is a bit slow, abort any uncompleted requests.
+    abortRef?.current?.abort();
+    abortRef.current = new AbortController();
```
Comment (MikeKingdom) [inline src/js/components/Data/stories/StarWars.js:37]: Cool.

**Review by MikeKingdom — APPROVED** [review]
Nice improvements. LGTM

File: src/js/components/DataSearch/DataSearch.js [hunk 2]
```diff
@@ -19,12 +24,15 @@ export const DataSearch = ({ ...rest }) => {
       id={`${dataId}--search`}
       name="_search"
       icon={<Search />}
+      reverse={addSubmit}
```
Comment (taysea) [inline src/js/components/DataSearch/DataSearch.js:27]: I think I was expecting the search icon to always be at the start, regardless of if the data submits on change or on submit. Having it on the left seems like it aids quick identification of the search field

File: src/js/components/DataSearch/DataSearch.js [hunk 3]
```diff
@@ -19,12 +24,15 @@ export const DataSearch = ({ ...rest }) => {
...
       type="search"
       {...rest}
     />
   );
 
-  if (noForm) content = <DataForm footer={false}>{content}</DataForm>;
+  if (noForm) {
```
Comment (taysea) [inline src/js/components/DataSearch/DataSearch.js:33]: Any reason this line changed? Seems like we can keep in the simplified format from before.

Comment (ericsoderberghp) [reply src/js/components/DataSearch/DataSearch.js]: reverted to simplified syntax

File: src/js/components/DataSearch/DataSearch.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/DataSearch/DataSearch.js]: Changed to always be at the start. I think I misinterpreted one of your earlier messages.

**Review by taysea — APPROVED** [review]
LGTM!

**Pull request merged**